vault/resource_identity_group.go

```diff
@@ -249,23 +249,28 @@ func identityGroupIDPath(id string) string {
 	return fmt.Sprintf("%s/id/%s", identityGroupPath, id)
 }
 
-func readIdentityGroupPolicies(client *api.Client, groupId string) ([]interface{}, error) {
-	var presentPolicies []interface{}
-	if resp, err := readIdentityGroup(client, groupId); err != nil {
-		return nil, fmt.Errorf("error reading IdentityGroup policies %q: %s", groupId, err)
-	} else {
-		presentPolicies = resp.Data["policies"].([]interface{})
+func readIdentityGroupPolicies(client *api.Client, groupID string) ([]interface{}, error) {
+	resp, err := readIdentityGroup(client, groupID)
+	if err != nil {
+		return nil, err
+	}
+	if resp == nil {
+		return nil, fmt.Errorf("error reading IdentityGroup does not exist: %s", groupID)
 	}
-	return presentPolicies, nil
+
+	if v, ok := resp.Data["policies"]; ok && v != nil {
+		return v.([]interface{}), nil
+	}
+	return make([]interface{}, 0), nil
 }
 
-func readIdentityGroup(client *api.Client, groupId string) (*api.Secret, error) {
-	path := identityGroupIDPath(groupId)
-	log.Printf("[DEBUG] Reading IdentityGroup %s from %q", groupId, path)
+func readIdentityGroup(client *api.Client, groupID string) (*api.Secret, error) {
```

Comment:
Might be nice to add a comment just above this function letting folks know that the resp can be nil.

Reply:
Done.